OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -2029,10 +2032,10 @@ public static ProcStatus Rgb(DMProcState state) {
                 return ProcStatus.Continue;
             }
 
-            int color1Value = (int)color1.UnsafeGetValueAsFloat();
-            int color2Value = (int)color2.UnsafeGetValueAsFloat();
-            int color3Value = (int)color3.UnsafeGetValueAsFloat();
-            byte aValue = (byte)Math.Clamp((int)a.UnsafeGetValueAsFloat(), 0, 255);
+            float color1Value = color1.UnsafeGetValueAsFloat();
+            float color2Value = color2.UnsafeGetValueAsFloat();
+            float color3Value = color3.UnsafeGetValueAsFloat();
+            byte aValue = a.IsNull ? (byte)255 : (byte)Math.Clamp((int)a.UnsafeGetValueAsFloat(), 0, 255);
```

Comment:
Should the int cast on `a` be removed here as well?

Reply:
I don't think so, pretty sure alpha is always a byte.